arkouda/categorical.py

```diff
@@ -1104,11 +1099,21 @@ def register(self, user_defined_name: str) -> Categorical:
         Objects registered with the server are immune to deletion until
         they are unregistered.
         """
-        [
-            p.register(f"{user_defined_name}.{n}")
-            for n, p in Categorical._get_components_dict(self).items()
-        ]
-        self.name = user_defined_name
+        if self.registered_name is not None and self.is_registered():
+            raise RegistrationError(f"This object is already registered as {self.registered_name}")
```

Comment:
I was thinking that this makes more sense because speaking for myself, I would definitely forget what I have registered. Maybe in the future we add a replace/update method that allows the overwrite.